tests/publiccloud/aistack_rbac_run.pm

```diff
@@ -0,0 +1,70 @@
+use Mojo::Base 'publiccloud::basetest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use publiccloud::utils;
+use version_utils;
+
+sub run_python_script {
+    my $script = shift;
+    my $logfile = "output.txt";
+    record_info($script, "Running python script: $script");
+    assert_script_run("$script");
+    assert_script_run("./$script 2>&1 | tee $logfile");
+}
+
+sub run {
+    my ($self, $args) = @_;
+
+    my $instance = $self->{my_instance} = $args->{my_instance};
+    my $provider = $self->{provider} = $args->{my_provider};
+
+    # Get Open WebUI ip address, add it to /etc/hosts, and verify connectivity
+    my $ipaddr = get_var('OPENWEBUI_IP');
+    my $host_name = get_var('OPENWEBUI_HOSTNAME');
+    record_info("debug $ipaddr");
+    assert_script_run("echo \"$ipaddr $host_name\" | sudo tee -a /etc/hosts > /dev/null");
+    record_info("Added $ipaddr to /etc/hosts with hostname $host_name");
+    assert_script_run("cat /etc/hosts");
+
+    my $curl_cmd = "curl -v -k https://$host_name";
+    my $curl_result = script_run($curl_cmd);
+    if ($curl_result == 0) {
+        record_info("Successfully connected to the open-webui service at $curl_cmd \n");
+    } else {
+        die "Unable to connect to the open-webui service at $curl_cmd\n";
+    }
+
+    # Get Open WebUI tests and admin credentials
+    my $rbac_tests_url = data_url("aistack/open-webui-rbac-tests.tar.gz");
+    my $admin_email = get_var('OPENWEBUI_ADMIN_EMAIL');
+    my $admin_password = get_var('OPENWEBUI_ADMIN_PWD');
+    record_info("Got Open WebUI admin credentials: $admin_email and $admin_password");
+
+    # Prepare and call tests
+    zypper_call("in python311");
+    assert_script_run("curl -O " . $rbac_tests_url);
+    assert_script_run("mkdir -p open-webui-rbac-tests && tar -xzvf open-webui-rbac-tests.tar.gz -C open-webui-rbac-tests && cd open-webui-rbac-tests");
+    assert_script_run("python3.11 -m venv myenv");
+    assert_script_run("source myenv/bin/activate");
+    assert_script_run("pip3 install -r requirements.txt");
+    assert_script_run("cp .env.example .env");
+    assert_script_run("export EMAIL='" . $admin_email . "' PASSWORD='" . $admin_password . "'");
+    record_info("Set env variables EMAIL = $admin_email and PASSWORD = $admin_password to be used in tests");
+    assert_script_run("pytest -vv --ENV remote tests");
+}
+
+sub post_fail_hook {
+    my $self = shift;
+    $self->cleanup();
+    $self->SUPER::post_fail_hook;
+}
+
+sub post_run_hook {
+    my $self = shift;
+    $self->cleanup();
+    $self->SUPER::post_run_hook;
+}
```

Comment:
Is this necessary? I don't see subroutine `cleanup` defined in this test module.